weblate/addons/forms.py

```diff
@@ -183,8 +183,20 @@ def __init__(self, *args, **kwargs):
 
 class JSONCustomizeForm(BaseAddonForm):
     sort_keys = forms.BooleanField(label=_("Sort JSON keys"), required=False)
-    indent = forms.IntegerField(
-        label=_("JSON indentation"), min_value=0, initial=4, required=True
+    indent_type = forms.ChoiceField(
+        label=_("JSON indentation"),
+        widget=forms.RadioSelect,
+        choices=(
+            ("integer", _("Treat Indentation as an integer")),
+            ("string", _("Treat Indentation as a string")),
+        ),
+        initial="integer",
+        required=True,
+    )
+    indent = forms.CharField(
+        label=_("JSON indentation"),
+        required=True,
+        initial="4",
```

Comment:
I don't think this is a good idea:

* It will be hard for most user to enter tab character here. Also, even when entered it will be invisible later when displaying the configuration.
* It no longer validates the integers, so the add-on will crash in case an invalid value is provided for the `integer` selection above.

The better approach could be to make the choice to define the actual indentation style (spaces / tabs) and the number only identifies number of them.

Reply:
Well JSON.dump says to use \t for tab characters, not actual tabs
It also allows you to make indentations based on whatever you choose.

Reply:
Yes, "\t" is a tab char as written in Python code. You don't want a literal `\t` there, it would end up as:

```
{
\t"foo": "bar"
}
```

Also, this clearly shows that you have not tested this patch at all...